sdk/cosmosdb/cosmos/src/client/V2ChangeFeed/ChangeFeedEpkRange.ts

```diff
@@ -0,0 +1,327 @@
+    }
+
+    this.queue = new FeedRangeQueue<ChangeFeedRange>();
+    this.continuationTokenMap = new ContinuationTokenMap(this.rId);
+
+    if (this.continuationToken) {
+      if (!this.continuationTokenRidMatchContainerRid()) {
+        throw new ErrorResponse("Container rid does not match with the continuation token rid");
+      } else {
+        for (const contToken of this.continuationToken.Continuation) {
+          const feedRange: ChangeFeedRange = new ChangeFeedRange(
+            contToken.minInclusive,
+            contToken.maxExclusive,
+            contToken.continuationToken
+          );
+
+          this.queue.enqueue(feedRange);
+          this.continuationTokenMap.addFeedRange(feedRange);
+        }
+      }
+    }
+  }
+
+  private continuationTokenRidMatchContainerRid(): boolean {
+    if (this.continuationToken.rid !== this.rId) {
+      return false;
+    }
+    return true;
+  }
+
+  public async fetchAllFeedRanges(): Promise<void> {
+    try {
+      const pkRanges = (
+        await this.partitionKeyRangeCache.onCollectionRoutingMap(this.url)
+      ).getOrderedParitionKeyRanges();
+      for (const pkRange of pkRanges) {
+        const feedRange: ChangeFeedRange = new ChangeFeedRange(
+          pkRange.minInclusive,
+          pkRange.maxExclusive,
+          ""
+        );
+        this.queue.enqueue(feedRange);
+        this.continuationTokenMap.addFeedRange(feedRange);
+      }
+    } catch (err) {
+      throw new Error(err.message);
+    }
+  }
+
+  public async fetchOverLappingFeedRanges(pkRange: PartitionKeyRange): Promise<void> {
+    const queryRange = new QueryRange(pkRange.minInclusive, pkRange.maxExclusive, true, false);
+    try {
+      const overLappingRanges = await this.partitionKeyRangeCache.getOverlappingRanges(
+        this.url,
+        queryRange
+      );
+
+      for (const overLappingRange of overLappingRanges) {
+        const feedRange: ChangeFeedRange = new ChangeFeedRange(
+          overLappingRange.minInclusive,
+          overLappingRange.maxExclusive,
+          ""
+        );
+        this.queue.enqueue(feedRange);
+        this.continuationTokenMap.addFeedRange(feedRange);
+      }
+    } catch (err) {
+      throw new ErrorResponse(err.message);
+    }
+  }
+  /**
+   * Gets a value indicating whether there are potentially additional results that can be retrieved.
+   *
+   * Initially returns true. This value is set based on whether there are any remaining unread epkRanges left in the queue.
+   *
+   * @returns Boolean value representing if whether there are potentially additional results that can be retrieved.
+   */
+  get hasMoreResults(): boolean {
+    return !this.queue.isEmpty();
+  }
+
+  /**
+   * Gets an async iterator which will yield pages of results from Azure Cosmos DB.
+   */
+  public async *getAsyncIterator(): AsyncIterable<ChangeFeedResponseV2<Array<T & Resource>>> {
+    do {
+      const result = await this.fetchNext();
+      if (result.count > 0) {
```

Comment:
How is a HTTP 304 (no more changes) communicated to the caller? (The NotModified check in https://learn.microsoft.com/en-us/azure/cosmos-db/nosql/change-feed-pull-model?tabs=dotnet#consuming-an-entire-containers-changes)